### ISSUE TITLE
How to visualize both a Point Cloud and Octomap?

### ISSUE THREAD

**Issue by Levi-Armstrong** [ignitionrobotics/ign-rendering#91]
I am interested to learn how to visualize a Point Cloud and Octomap in Ignition Rendering. I am currently using a depth sensor in ignition to generating a point cloud and I currentlypublish it for visualization in Rviz. I have an application built using Ignition GUI and if possible I would like to visualize the information in the ignition scene.

**Comment by Levi-Armstrong** [comment]
It looks like I may be able to leverage the scene marker which supports points, but need to test. Still looking into how best to visualize an Octomap.

**Comment by iche033** [comment]
for now you may be able to visualize point clouds by publishing to the marker topic like in the [marker example](https://github.com/ignitionrobotics/ign-gazebo/tree/master/examples/standalone/marker) in ign-gazebo.

We just started a GOSC project to create a lidar visual (issue #84) in ign-rendering and may extend to point cloud visualization if there is time in the end, cc @mihirk284

**Comment by Levi-Armstrong** [comment]
I will hold of on the point cloud visual and track issue #84 to avoid duplicating work. 

Though I could look at the Octomap. Would your preferred approach be similar to issue #84 for the octomap? I could begin working on an OctomapVisual.

**Comment by iche033** [comment]
> Though I could look at the Octomap. Would your preferred approach be similar to issue #84 for the octomap? I could begin working on an OctomapVisual.

yeah that'd would be great, thanks! I have not looked at rviz's implementation to see how it's done but it looks the visualization just consists of colored cubes. Hopefully it'll be straightforward.

**Comment by Levi-Armstrong** [comment]
> yeah that'd would be great, thanks! I have not looked at rviz's implementation to see how it's done but it looks the visualization just consists of colored cubes. Hopefully it'll be straightforward.

That is what I am thinking. I hope I can use what is in rviz and integrating into the ogre1 and ogre2.

**Comment by Levi-Armstrong** [comment]
I just now getting back to revisiting this. @iche033 Has any progress been made on visualizing point clouds or octomaps?

**Comment by iche033** [comment]
We now have a [LidarVisual](https://github.com/gazebosim/gz-rendering/blob/gz-rendering7/include/gz/rendering/LidarVisual.hh) class for visualizing lidar point clouds. The ogre2 implementation is here: https://github.com/gazebosim/gz-rendering/blob/gz-rendering7/ogre2/src/Ogre2LidarVisual.cc.

There is a gui plugin in gz-sim that essentially subscribes to lidar data to populates the LidarVisual:
https://github.com/gazebosim/gz-sim/tree/gz-sim7/src/gui/plugins/visualize_lidar
